From 835714082d6bd427e2b820d245b1fe9cd426de40 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 11 Apr 2022 13:26:52 +0200
Subject: [PATCH 155/954] refactor(st): update DDR mapping

The first 32MB of DDR will be used for Cortex-M33. Then Secure OS from
32MB to 64MB. The rest of the DDR will be used for non-secure OS.
STM32MP_BL33_BASE is then upated in the platform code, and the fw-config
file is modified to take those areas into account.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 440fdb77eddcb206576be0098116e1894bb5b152 refactor(st): update DDR mapping]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I4c593d3485766023f0567c197f2b0540bd3af079
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/247648
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 fdts/stm32mp25-fw-config.dtsi | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/fdts/stm32mp25-fw-config.dtsi b/fdts/stm32mp25-fw-config.dtsi
index f55a341083..17d8f70391 100644
--- a/fdts/stm32mp25-fw-config.dtsi
+++ b/fdts/stm32mp25-fw-config.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
 /*
- * Copyright (c) 2021, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2021-2022, STMicroelectronics - All Rights Reserved
  */
 
 #include <common/tbbr/tbbr_img_def.h>
@@ -11,12 +11,10 @@
 #error "DDR_SIZE is not defined"
 #endif
 
-#define DDR_NS_BASE	STM32MP_DDR_BASE
-#define DDR_SEC_SIZE	0x01000000
-#define DDR_SEC_BASE	(STM32MP_DDR_BASE + (DDR_SIZE - DDR_SEC_SIZE))
-#define DDR_SHARE_SIZE	0x00200000
-#define DDR_SHARE_BASE	(DDR_SEC_BASE - DDR_SHARE_SIZE)
-#define DDR_NS_SIZE	(DDR_SHARE_BASE - DDR_NS_BASE)
+#define DDR_SEC_BASE	(STM32MP_DDR_BASE + 0x02000000)
+#define DDR_SEC_SIZE	0x02000000
+#define DDR_NS_BASE	(DDR_SEC_BASE + DDR_SEC_SIZE)
+#define DDR_NS_SIZE	(STM32MP_DDR_BASE + DDR_SIZE - DDR_NS_BASE)
 
 /dts-v1/;
 
-- 
2.53.0

